File: leetcode/golang/LCR_175_er_cha_shu_de_shen_du_lcof.go
type TreeNode struct {
         Val int
         Left *TreeNode
         Right *TreeNode
}

func calculateDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    res := 0
    var queue = []*TreeNode {root}
    for len(queue) != 0 {
        res++
        for i := len(queue); i > 0; i-- {
            node := queue[0]
            queue = queue[1:]
            if node.Left != nil {
                queue = append(queue, node.Left)
            }
            if node.Right != nil {
                queue = append(queue, node.Right)
            }
        } 
    }
    return res
}

File: leetcode/c++/LCR_175_er_cha_shu_de_shen_du_lcof.cpp
// https://leetcode.cn/problems/er-cha-shu-de-shen-du-lcof/description

#include <bits/stdc++.h>
using namespace std;

struct TreeNode {
      int val;
      TreeNode *left;
      TreeNode *right;
      TreeNode() : val(0), left(nullptr), right(nullptr) {}
      TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
      TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int calculateDepth(TreeNode* root) {
        if (root == nullptr)
            return 0;
        return max(calculateDepth(root->left), calculateDepth(root->right)) + 1;
    }
};

// DFS，时间复杂度为O(n)，空间复杂度为O(n)，没有BFS快